sdks/swift/Mentat/MentatTests/MentatTests.swift

```diff
@@ -0,0 +1,836 @@
+                [:db/add "u" :db/cardinality :db.cardinality/one]
+            ]
+            """
+            let report = try mentat.transact(transaction: schema)
+            let stringEntid = report.entidForTmpId(tmpId: "s")!
+
+            let data = """
+            [
+                [:db/add "a" :foo/boolean true]
+                [:db/add "a" :foo/long 25]
+                [:db/add "a" :foo/instant #inst "2017-01-01T11:00:00.000Z"]
+                [:db/add "a" :foo/double 11.23]
+                [:db/add "a" :foo/string "The higher we soar the smaller we appear to those who cannot fly."]
+                [:db/add "a" :foo/keyword :foo/string]
+                [:db/add "a" :foo/uuid #uuid "550e8400-e29b-41d4-a716-446655440000"]
+                [:db/add "b" :foo/boolean false]
+                [:db/add "b" :foo/ref \(stringEntid)]
+                [:db/add "b" :foo/long 50]
+                [:db/add "b" :foo/instant #inst "2018-01-01T11:00:00.000Z"]
+                [:db/add "b" :foo/double 22.46]
+                [:db/add "b" :foo/string "Silence is worse; all truths that are kept silent become poisonous."]
+                [:db/add "b" :foo/uuid #uuid "4cb3f828-752d-497a-90c9-b1fd516d5644"]
+            ]
+            """
+            return try mentat.transact(transaction: data)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+        return nil
+    }
+
+    func test1TransactVocabulary() {
+        do {
+            let mentat = Mentat()
+            let report = try transactSchema(mentat: mentat)
+            XCTAssertNotNil(report)
+            assert(report.txId > 0)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+    }
+
+    func test2TransactEntities() {
+        do {
+            let mentat = Mentat()
+            let _ = try self.transactSchema(mentat: mentat)
+            let report = try self.transactEdn(mentat: mentat)
+            XCTAssertNotNil(report)
+            assert(report.txId > 0)
+            let entid = report.entidForTmpId(tmpId: "a17592186045438")
+            assert(entid == 65566)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+    }
+
+    func testQueryScalar() {
+        let mentat = newStore()
+        let query = "[:find ?n . :in ?name :where [(fulltext $ :community/name ?name) [[?e ?n]]]]"
+        let expect = expectation(description: "Query is executed")
+        try! mentat.query(query: query).bind(varName: "?name", toString: "Wallingford").executeScalar(callback: { (scalarResult, error) in
+            assert(error == nil, "Unexpected error: \(String(describing: error))")
+            guard let result = scalarResult?.asString() else {
+                return assertionFailure("No String value received")
+            }
+            assert(result == "KOMO Communities - Wallingford")
+            expect.fulfill()
+        })
+        waitForExpectations(timeout: 1) { error in
```

Comment:
I don't much like expectations; they were unreliable in Firefox tests. I would prefer to give the caller control over which queue runs a query, so they can kick off a background task and run a bunch of queries in a row… and that allows us to write serial tests, too.